src/AppInstallerCLICore/Commands/UninstallCommand.cpp

```diff
@@ -21,12 +21,15 @@ namespace AppInstaller::CLI
             Argument::ForType(Args::Type::Id),
             Argument::ForType(Args::Type::Name),
             Argument::ForType(Args::Type::Moniker),
+            Argument::ForType(Args::Type::ProductCode),
             Argument::ForType(Args::Type::Version),
             Argument::ForType(Args::Type::Channel),
             Argument::ForType(Args::Type::Source),
             Argument::ForType(Args::Type::Exact),
             Argument::ForType(Args::Type::Interactive),
             Argument::ForType(Args::Type::Silent),
+            Argument::ForType(Args::Type::Purge),
+            Argument::ForType(Args::Type::Preserve),
```

Comment:
Do these need to be added to the UpgradeCommand as well? I could see a case where a user would want to `--preserve` any user data files inside their folder when upgrading a portable app

Reply:
Even if the default is to preserve the directory, I could then see a case for a user to want to use —purge to perform a clean upgrade

Reply:
I decided to only add `--purge` to the UpgradeCommand and not have the `UninstallBehavior.PurgePortablePackage` settings apply to the upgrade command as that would seem like an unexpected behavior. `--purge` will now remove all files and directories when performing an `uninstallPrevious`